src/content/peerconnection/negotiate-timing/js/main.js

```diff
@@ -193,29 +233,47 @@ async function getAudioImpairment(audioTransceiver) {
 }
 
 async function baselineAudioImpairment(pc) {
-  audioTransceiver = pc.getTransceivers().find(tr => tr.receiver.track.kind == 'audio');
-  console.log('Found audio transceiver');
-  audioImpairmentAtStart = await getAudioImpairment(audioTransceiver);
+  audioReceiver = pc.getReceivers().find(r => r.track.kind == 'audio');
+  console.log('Found audio receiver');
+  audioImpairmentAtStart = await getAudioImpairment(audioReceiver);
 }
 
 async function measureAudioImpairment(pc) {
   const startTime = performance.now();
-  const audioImpairmentNow = await getAudioImpairment(audioTransceiver);
+  const audioImpairmentNow = await getAudioImpairment(audioReceiver);
   console.log('Measurement took ' + (performance.now() - startTime) + ' msec');
   return audioImpairmentNow - audioImpairmentAtStart;
 }
 
 
 async function renegotiate() {
   renegotiateButton.disabled = true;
-  adjustTransceiverCounts(pc1, parseInt(videoSectionsField.value));
+  let targetVideoSections = parseInt(videoSectionsField.value);
+  if (sdpSemantics == "unified-plan") {
+    adjustTransceiverCounts(pc1, targetVideoSections);
+  } else {
+    await adjustSenderCounts(pc1, targetVideoSections);
+  }
   await baselineAudioImpairment(pc2);
-  const previousVideoTransceiverCount = pc2.getTransceivers().filter(tr => tr.receiver.track.kind == 'video').length;
+  const previousVideoTransceiverCount = pc2.getReceivers().filter(r => r.track.kind == 'video').length;
   result = await runOfferAnswer();
+  if (sdpSemantics == "plan-b") {
+    // Clean up temporary clones created inside of adjustSenderCounts() above.
+    // This includes all senders except the first audio and video ones used for
+    // the preview.
+    const additionalSenders = pc1.getSenders().filter(
+        s => s!= videoSender && s != audioSender);
+    for (let i = 0; i < additionalSenders.length; i++) {
+      if (additionalSenders[i].track) {
+        additionalSenders[i].track.stop();
+        await additionalSenders[i].replaceTrack(null);
+      }
+    }
+  }
   console.log(`Renegotiate finished after ${result.elapsedTime} milliseconds`);
-  const currentVideoTransceiverCount = pc2.getTransceivers().filter(tr => tr.receiver.track.kind == 'video').length;
+  const currentVideoTransceiverCount = pc2.getReceivers().filter(r => r.track.kind == 'video').length;
   result.audioImpairment = await measureAudioImpairment(pc2);
```

Comment:
can you move the measurement before the clone cleanup, so that we don't get hit by any audio impact from that part?